src/main/java/com/epam/lab/payments/web/AuthorizationController.java

```diff
@@ -0,0 +1,67 @@
+package com.epam.lab.payments.web;
+
+import com.epam.lab.payments.UserValidator;
+import com.epam.lab.payments.domain.UserEntity;
+import com.epam.lab.payments.services.AuthorizationService;
+import com.epam.lab.payments.services.SecurityService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.MultiValueMap;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.validation.Valid;
+import java.util.HashMap;
+
+@Controller
+@RequiredArgsConstructor
+public class AuthorizationController {
+    private final AuthorizationService authorizationService;
+    private final SecurityService securityService;
+    private final UserValidator userValidator;
+
+    @RequestMapping(value="/registration", method = RequestMethod.GET)
+    public ModelAndView registration(){
+        ModelAndView modelAndView = new ModelAndView();
+        UserEntity user = new UserEntity();
+        modelAndView.addObject("user", user);
+        modelAndView.setViewName("registration");
+        return modelAndView;
+    }
+
+    @RequestMapping(value = "/registration", method = RequestMethod.POST)
+    public ModelAndView createNewUser(@Valid UserEntity user, BindingResult bindingResult) {
+        ModelAndView modelAndView = new ModelAndView();
+        UserEntity userExists = authorizationService.findUserByEmail(user.getEmail());
+        if (userExists != null) {
+            bindingResult
+                    .rejectValue("email", "error.user",
+                            "There is already a user registered with the email provided");
+        }
+        if (bindingResult.hasErrors()) {
+            modelAndView.setViewName("registration");
```

Comment:
Define constant instead of duplicating "registration" several times.

Reply:
done